### ISSUE TITLE
Filter using Adminhtml/Block/Widget/Grid/Column/Filter/Select.php

### ISSUE THREAD

**Issue by pljspahn** [kalenjordan/custom-reports#20]
I added a field to filter to my report, and used the `select` type of filter.

I am not seeing values in the `select` field, so I am guessing there is no source model added ... I read through some of the issues here, but and I saw some discussion on it, but I am not clear on whether this should be working or not. Is filtering by select types supported?

**Comment by kalenjordan** [comment]
Good grief it's been a while and I don't remember hardly any of this stuff.  @rgranadino any thoughts?

**Comment by rgranadino** [comment]
I don't think this is possible at the moment. It probably wouldn't be too much effort to add in tho. The tricky parts is that the current JSON structure is a simple key/value pair and it would need to be expanded on and remain backwards compatible. 

The workable structure might be something like:

    {"filterable":
        { 
          "order_status": {"type": "widget_grid_column_filter_select", "options": ["option 1", "option2"]}
      }
    } 

The `Clean_SqlReports_Block_Adminhtml_Report_View_Grid::_prepareColumns` method would need some refactoring to check that the value is in array and not a string. If it's an array, then set the type and add the options on the column Array definition passed to the `addColumn` call.

**Comment by sreichel** [comment]
How about ...

```
{"filterable":
    { 
      "order_status": {"type": "widget_grid_column_filter_select", "source_model": "catalog/product_status", "method": "yourMehtod"}
  }
} 
```

... where `method` could default to toOptionArray() ...

**Comment by hajota** [comment]
I defined a select filter on a column as following.
`"Verkaufskanal": "adminhtml/widget_grid_column_filter_select"`
The table shows a select for filtering, but it has no options. The column is queried as following.
`if(sfoi.store_id in (13,14,29), 'B2B', 'B2C') as Verkaufskanal`
How is the correct syntax for a select filter showing select options?
